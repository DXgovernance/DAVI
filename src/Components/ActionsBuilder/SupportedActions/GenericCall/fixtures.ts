import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'Components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const genericDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.GENERIC_CALL,
  function: ERC20Contract.getFunction('increaseAllowance'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    param1: BigNumber.from(10000),
    param2: BigNumber.from(10000),
    param3: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
