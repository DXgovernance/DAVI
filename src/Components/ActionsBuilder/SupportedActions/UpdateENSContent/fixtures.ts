import { BigNumber, utils } from 'ethers';
import { DecodedCall } from 'Components/ActionsBuilder/types';
import { SupportedAction } from 'Components/ActionsBuilder/types';
import ENSPublicResolver from 'contracts/ENSPublicResolver.json';

const ENSPublicResolverContract = new utils.Interface(ENSPublicResolver.abi);

export const mockDecodedCallUpdateENSContent: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ENS_UPDATE_CONTENT,
  function: ENSPublicResolverContract.getFunction('setContenthash'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    node: '0x0000000000000000000000000000000000000000',
    hash: '0x0000000000000000000000000000000000000000',
  },
  optionalProps: {
    ensName: 'test',
    ipfsHash: 'Qm',
  },
};
