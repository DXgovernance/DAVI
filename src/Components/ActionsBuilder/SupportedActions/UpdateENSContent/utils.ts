import namehash from 'eth-ens-namehash';
import contenthash from 'content-hash';
import {
  MAINNET_ID,
  LOCALHOST_ID,
  GOERLI_ID,
  ARBITRUM_ID,
  ARBITRUM_TESTNET_ID,
} from 'utils';

export const convertToNameHash = (name: string) => namehash.hash(name);

export const convertToContentHash = (ipfsHash: string) =>
  contenthash.fromIpfs(ipfsHash);

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const getBlockChainUrl = (chainId: number, address: string) => {
  switch (chainId) {
    case LOCALHOST_ID:
    case MAINNET_ID:
      return `https://etherscan.io/address/${address}`;
    case ARBITRUM_ID:
      return `https://arbitrum.io/address/${address}`;
    case ARBITRUM_TESTNET_ID:
      return `https://arbitrum.io/testnet/address/${address}`;
    case GOERLI_ID:
      return `https://goerli.etherscan.io/address/${address}`;
    default:
      throw Error(`The chainId or address is not supported`);
  }
};
