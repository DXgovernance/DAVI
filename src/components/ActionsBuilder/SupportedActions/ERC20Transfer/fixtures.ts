import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const erc20TransferEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const erc20TransferDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    _to: '0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698',
    _value: BigNumber.from('5000000000000000000'),
  },
};
