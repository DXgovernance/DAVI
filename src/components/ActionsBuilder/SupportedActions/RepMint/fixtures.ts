import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber, utils } from 'ethers';
import ERC20SnapshotRep from 'contracts/ERC20SnapshotRep.json';

const ERC20SnapshotRepContract = new utils.Interface(ERC20SnapshotRep.abi);

export const repMintEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.REP_MINT,
  function: ERC20SnapshotRepContract.getFunction('mint'),
  to: '',
  value: BigNumber.from(0),
  args: {
    to: '',
    amount: BigNumber.from(0),
  },
};

export const repMintDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.REP_MINT,
  function: ERC20SnapshotRepContract.getFunction('mint'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
    amount: BigNumber.from(10000),
  },
};
