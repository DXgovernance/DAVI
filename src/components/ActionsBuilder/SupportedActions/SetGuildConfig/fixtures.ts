import { BigNumber, utils } from 'ethers';
import { SupportedAction } from 'components/ActionsBuilder/types';
import BaseERC20Guild from 'contracts/BaseERC20Guild.json';

export const mockBigNumber = BigNumber.from(100000000);
export const BaseERC20GuildContract = new utils.Interface(BaseERC20Guild.abi);

export const emptyDecodedCall = {
  from: '',
  callType: SupportedAction.SET_GUILD_CONFIG,
  function: BaseERC20GuildContract.getFunction('setConfig'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _proposalTime: '',
    _timeForExecution: '',
    _votingPowerPercentageForProposalExecution: '',
    _votingPowerPercentageForProposalCreation: '',
    _voteGas: '',
    _maxGasPrice: '',
    _maxActiveProposals: '',
    _lockTime: '',
    _minimumMembersForProposalCreation: '',
    _minimumTokensLockedForProposalCreation: '',
  },
  optionalProps: {},
};
