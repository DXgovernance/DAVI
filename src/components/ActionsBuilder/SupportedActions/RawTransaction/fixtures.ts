import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber } from 'ethers';
import { MOCK_ADDRESS } from 'hooks/Guilds/ens/fixtures';

export const rawDataCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.RAW_TRANSACTION,
  function: null,
  to: MOCK_ADDRESS,
  value: BigNumber.from('1000000000000000000'),
  args: null,
  optionalProps: {
    data: '0x01',
  },
};
