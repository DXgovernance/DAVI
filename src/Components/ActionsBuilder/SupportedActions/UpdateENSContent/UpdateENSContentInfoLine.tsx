import { Segment } from '../common/infoLine';
import { MdArrowRightAlt } from 'react-icons/md';
import ENSIcon from 'assets/images/ens.svg';
import { useUpdateEnsContent } from 'hooks/Guilds/guild/useUpdateEnsContent';
import { StyledENSIcon } from './styles';
import useENSAvatar from 'hooks/Guilds/ether-swr/ens/useENSAvatar';
import { MAINNET_ID } from 'utils';
import Avatar from 'old-components/Guilds/Avatar';
import { shortenAddress } from 'utils';
import { ActionViewProps } from '..';
import { useTranslation } from 'react-i18next';

const UpdateENSContentInfoLine: React.FC<ActionViewProps> = ({
  decodedCall,
  compact,
}) => {
  const { parsedData } = useUpdateEnsContent({ decodedCall });
  const { ensName, imageUrl } = useENSAvatar(parsedData?.from, MAINNET_ID);
  const { t } = useTranslation();

  return (
    <>
      <Segment>
        <StyledENSIcon src={ENSIcon} />
      </Segment>
      <Segment>{!compact ? t('ens.updateContent') : ''}</Segment>
      <Segment>
        <MdArrowRightAlt />
      </Segment>
      <Segment>
        <Avatar
          defaultSeed={parsedData?.from}
          src={imageUrl}
          size={compact ? 14 : 24}
        />
      </Segment>
      <Segment>
        {ensName || shortenAddress(parsedData?.from, compact ? 2 : 4)}
      </Segment>
    </>
  );
};

export default UpdateENSContentInfoLine;
