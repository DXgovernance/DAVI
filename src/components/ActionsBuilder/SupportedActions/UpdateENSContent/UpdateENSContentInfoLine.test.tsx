import { render } from 'utils/tests';
import UpdateENSContentInfoLine from './UpdateENSContentInfoLine';
import { mockDecodedCallUpdateENSContent } from './fixtures';

jest.mock('hooks/Guilds/ether-swr/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    avatarUri: 'test',
    imageUrl: 'test',
    ensName: 'test.eth',
  }),
}));

describe('UpdateENSInfoLine', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <UpdateENSContentInfoLine decodedCall={mockDecodedCallUpdateENSContent} />
    );
    expect(container).toMatchSnapshot();
  });
});
