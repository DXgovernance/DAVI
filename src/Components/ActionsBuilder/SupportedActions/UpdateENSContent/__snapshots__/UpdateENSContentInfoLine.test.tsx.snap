// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateENSInfoLine Should match snapshot 1`] = `
<div>
   
  <span
    class="sc-bdfBQB enldmJ"
  >
    <div
      class="sc-dlfnuX fjXZjO sc-iqHYmW CgZEx"
    >
      <img
        src="ens.svg"
      />
    </div>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    ens.updateContent
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M16.01 11H4v2h12.01v3L20 12l-3.99-4z"
      />
    </svg>
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    <img
      alt="Avatar"
      class="sc-fodVek sc-fFucqa kyFjTL gPqAoJ"
      data-testid="avatar"
      size="24"
      src="test"
    />
  </span>
  <span
    class="sc-bdfBQB enldmJ"
  >
    test.eth
  </span>
   
</div>
`;
